helmfile/values/calico-default-deny.yaml.gotmpl

```diff
@@ -0,0 +1,22 @@
+namespaceSelector:
+{{- if .Values.network.allowKubeSystem }}
+  has(projectcalico.org/name) && projectcalico.org/name not in {"kube-system"}
+{{- else }}
+  has(projectcalico.org/name)
+{{- end }}
+
+egressPolicies:
+{{- if .Values.network.allowDns }}
+  # allow all namespaces to communicate to DNS pods
+  - action: Allow
+    protocol: UDP
+    destination:
+      selector: 'k8s-app == "kube-dns"'
```

Comment:
The pods should communicate to node-local-dns and it as the label `k8s-app=node-local-dns`
I guess that node-local-dns then require to talk to coredns and then coredns out from the cluster.

Reply:
I'm not sure if this is true, since the pods don't know about `node-local-dns` but it rather intercepts traffic that has the destination `kube-dns`

Reply:
That can be so.

Reply:
In my testing with regular networkpolicies it did not necessarily work to just select the dns pods. Me and wallace decided to just accept UDP egress traffic to port 53 in general. Did you test that this seemed to work, otherwise just opening it up more might be better.

Reply:
Also, don't you need an ingress rule for the dns pods? So that they can accept the traffic

Reply:
From a safety perspective, I think opening it up more is a bad idea. If you can control DNS, then it's trivial to spoof the validity of certificates and thus very easy to do a MITM attack. So I would rather get this done correctly from the start tbh. although I would be okey to leave it more open if we create an issue to fix it later

Reply:
I took this from the [example in Calico's docs](https://projectcalico.docs.tigera.io/security/kubernetes-default-deny), I did not test it directly. I suppose the reason why it doesn't have ingress rules for the DNS pods is because it assumes you allow all traffic in kube-system.

That said, I saw that the harbor networkpolicy includes its own DNS rule. If we want to do this for all the different networkpolicies we're adding, then I suppose it would make more sense to just remove this option from the globalnetworkpolicy and handle it on a per-app basis. What do you think?

Reply:
That might actually make most sense, if we want to limit all apps as much as possible we might not even want to have dns enabled for some namespace (although I highly doubt it).

Reply:
I removed the option, we'll handle it on a per-app basis instead.